Library/Homebrew/test/Rakefile

```diff
@@ -1,8 +1,14 @@
 require "rake"
 require "rake/testtask"
 
+def mac?
+  /darwin/i === RUBY_PLATFORM && !ENV["HOMEBREW_TEST_GENERIC_OS"]
+end
+
 TEST_DIRECTORY = File.dirname(File.expand_path(__FILE__))
-TEST_FILES = Dir["#{TEST_DIRECTORY}/test_*.rb"]
+TEST_FILES = Dir["#{TEST_DIRECTORY}/test_*.rb"].reject do |f|
+  f.include?("/test_os_mac_") && !mac?
+end
```

Comment:
I realize you have already shuffled around quite a few test methods. Would you consider a more fine-grained approach to excluding certain tests when they are platform-specific?

I was thinking of something like `needs_os :mac` at the top of those tests (similar to other `needs_*` methods used in tests), that would cause the test to be skipped. One advantage would be that this will require a lot less shuffling around and the diff (at least in the `test/` directory) will be minimal. We had already discussed something like this with @sjackman on Slack.